### PR TITLE
Validate the sky.uk/allow annotation entries

### PR DESCRIPTION
**Pull request by alangibson01**
Each comma-separated entry should be either an IP address or a CIDR range. Ingresses where this isn't the case will be skipped, with an error message logged detailing the invalid entries.

### CODE REVIEW

File: controller/ingress_entry.go [hunk 1]
```diff
@@ -64,6 +66,22 @@ func (e IngressEntry) validate() error {
...
+			if _, _, err := net.ParseCIDR(allowEntry); err != nil {
+				invalidAllowEntries = append(invalidAllowEntries, allowEntry)
+			}
+		}
+	}
+
+	if len(invalidAllowEntries) == 1 {
+		return fmt.Errorf("invalid entry in sky.uk/allow: %s", invalidAllowEntries[0])
+	} else if len(invalidAllowEntries) > 1 {
+		return fmt.Errorf("invalid entries in sky.uk/allow: %s", strings.Join(invalidAllowEntries, ","))
```
Comment (sebbonnet) [inline controller/ingress_entry.go]: I think we should indicate which host the entry in invalid for  as there could be a large number of ingresses.

File: controller/ingress_entry.go [hunk 2]
```diff
@@ -64,6 +66,22 @@ func (e IngressEntry) validate() error {
...
+	for _, allowEntry := range e.Allow {
+		if net.ParseIP(allowEntry) == nil {
+			if _, _, err := net.ParseCIDR(allowEntry); err != nil {
+				invalidAllowEntries = append(invalidAllowEntries, allowEntry)
+			}
+		}
+	}
+
+	if len(invalidAllowEntries) == 1 {
+		return fmt.Errorf("invalid entry in sky.uk/allow: %s", invalidAllowEntries[0])
```
Comment (sebbonnet) [inline controller/ingress_entry.go]: Any reason to have this additional block other than to change the message from "entry" to "entries"? `strings.Join` won't print the comma if there is only one element

File: controller/ingress_entry.go (hunk 1 above)
Comment (sebbonnet) [reply controller/ingress_entry.go]: besides in when the entry is empty, it would be quite hard to find which one is missing

**Comment by sebbonnet** [comment]
You should also add a note in the CHANGELOG.md :)

File: controller/ingress_entry.go (hunk 1 above)
Comment (alangibson01) [reply controller/ingress_entry.go]: Actually, I ought to add a test that covers the case of empty entries as well.

File: controller/ingress_entry.go (hunk 2 above)
Comment (alangibson01) [reply controller/ingress_entry.go]: Yeah, just being picky grammatically, but there isn't much need for that. I'll simplify the code.

**Comment by alangibson01** [comment]
Changes made as requested.
